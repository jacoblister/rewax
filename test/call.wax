(func call_empty_test 
  (print "call_empty_test:")
)

(func call_int_test(param x int) (result int)
  (print "call_int_test:")
  (return x)
)

(func call_arr_test (param a (arr int)) (result (arr int))
  (print "call_arr_test:")
  (set a 0 4)
  (set a 1 5)
  (set a 2 6)
  (return a)
)

(func call_return_nothing
  (print "call_return_nothing:")
  (return)
)

(func main (result int)
  (call call_empty_test)
  (print (call call_int_test 2))
  
  (let a (arr int) (alloc (arr int) 1 2 3))
  (let r (arr int) (call call_arr_test a))
  (print (get a 0))
  (print (get a 1))
  (print (get a 2))
  (print (get r 0))
  (print (get r 1))
  (print (get r 2))

  (call call_return_nothing)

  (return 0)
)