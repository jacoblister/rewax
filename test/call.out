call_empty_test:
call_int_test:
2
call_arr_test:
4
5
6
4
5
6
call_return_nothing:
